functional_tests/driver_tests/test_record_task_execution.py

```diff
@@ -0,0 +1,30 @@
+import pytest
+from swimlane.exceptions import SwimlaneException
+
+@pytest.fixture(autouse=True, name='app', scope='module')
+def my_fixture(helpers):
+    # setup stuff
+    helpers.import_content('basic_task_app.ssp')
+    yield helpers.swimlane_instance.apps.get(name='Basic Task App')
+    # teardown stuff
+    helpers.cleanupData()
+
+
+def test_execute_task(app):
+    # sample_task has input 'Single-line text field' and ouput 'Text List'.
+    # Task splits input into a list and sends to output.
+    record = app.records.create()
+    record["Single-line text field"] = "Testing 12345"
+    record.execute_task("sample_task")  
+    assert record['Text List'][0] == 'Testing'
+    assert record['Text List'][1] == '12345'
+    record.delete()
+
+
+def test_execute_task_raises(app):
```

Comment:
Did it failed because it's not found or because it threw exception? Should we differentiate?

Reply:
It fails because the task is configured to fail and it throws an exception, and in turn, a failed status for a job throws an exception with the message body of the job.

https://github.com/swimlane/swimlane-python/blob/1f52642fbaf892565f695ff23b79d776f5297006/swimlane/core/resources/record.py#L445-L446

If the task is not found a ValueError Exception is thrown.
https://github.com/swimlane/swimlane-python/blob/1f52642fbaf892565f695ff23b79d776f5297006/swimlane/core/adapters/task.py#L17-L24

So to answer your question about should we differentiate, for those we are. Let me know if that answers the questions.

Reply:
It does, can we add the test for invalid name then?

Reply:
I have added a test for an invalid name.